code/datums/explosion.dm

```diff
@@ -119,11 +121,13 @@ GLOBAL_LIST_EMPTY(explosions)
 				// If inside the blast radius + world.view - 2
 				if(dist <= round(max_range + world.view - 2, 1))
 					M.playsound_local(epicenter, null, 100, 1, frequency, falloff = 5, S = explosion_sound)
+					shake_camera(M, 50, orig_max_distance - dist)
```

Comment:
5 seconds of shake MIGHT be a bitch much.

Also time of shake and amount of shake *might* need to scale together.

Reply:
also this might need a cap on it, as atm it doesn't scale so well when you take into account admin explosions.

Reply:
Yeah i need to fuck around and find some good numbers